### PR TITLE
Describe RFC process

### PR DESCRIPTION
**Pull request by oudod**

### CODE REVIEW

**Review by alexauroradev — COMMENTED** [review]
LGTM after adding necessary info in TODOs:

First TODO with examples: you can use the ones that we've discussed today.

- **Services isolation**. A major improvement which is aimed at separating storage and execution environments of services.
- **Transaction finality**. Adding additional mechanisms for transaction finalization without inclusion to a blockchain.
- **Code parallelization**. Make Exonum able to run on several cores simultaneously.

Second todo: probably need something similar for "motion for final comment period" as in Rust and also info about when the number is attached.

**Review by alexauroradev — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -1 +1,225 @@
-# Exonum RFCs
\ No newline at end of file
+# Exonum RFCs
+[Exonum RFCs]: #exonum-rfcs
+
+Exonum RFC process is partially derived from [the Rust RFC process](https://github.com/rust-lang/rfcs/blob/master/README.md)
```
Comment (slowli) [inline README.md]: I couldn't find out the license for the Rust RFCs. Are you sure we can use it?

File: README.md [hunk 2]
```diff
@@ -1 +1,225 @@
...
+## When you need to follow this process
+[When you need to follow this process]: #when-you-need-to-follow-this-process
+
+You need to follow this process if you intend to make "substantial" changes to
+Exonum core or the RFC process itself. What constitutes a
+"substantial" change is evolving based on community norms and varies depending
+on what part of the ecosystem you are proposing to change, but may include the
+following.
+
+- API changes.
```
Comment (slowli) [inline README.md]: Inconsistent indentation with other lists. Also: what does "API changes" entail? Changes in the public interface of the `exonum` crate?

File: README.md [hunk 3]
```diff
@@ -1 +1,225 @@
...
+
+## Reviewing RFCs
+[Reviewing RFCs]: #reviewing-rfcs
+
+While the RFC pull request is up, the shepherd may schedule meetings with the
+author and/or relevant stakeholders to discuss the issues in greater detail. In
+either case a summary from the meeting will be posted back to the RFC pull
+request.
+
+Maintainers of related repositories makes final decisions about RFCs after the
```
Comment (slowli) [inline README.md]: Nitpick: s/makes/make/

File: README.md [hunk 4]
```diff
@@ -1 +1,225 @@
...
+consensus and community norms, not impose more structure than necessary.
+
+## License
+
+The Exonum RFCs are licensed under the Creative Commons Non-Commercial
+Share-Alike International License (version 4.0).
+See [LICENSE](LICENSE) for details.
+
+[RFC issue tracker]: https://github.com/exonum/rfcs/issues
+[RFC repository]:  https://github.com/exonum/rfcs
```
Comment (slowli) [inline README.md]: Nitpick: unnecessary spacing.

File: README.md [hunk 5]
```diff
@@ -1 +1,225 @@
...
+[Help this is all too informal!]: #help-this-is-all-too-informal
+
+The process is intended to be as lightweight as reasonable for the present
+circumstances. As usual, we are trying to let the process be driven by
+consensus and community norms, not impose more structure than necessary.
+
+## License
+
+The Exonum RFCs are licensed under the Creative Commons Non-Commercial
+Share-Alike International License (version 4.0).
```
Comment (slowli) [inline README.md]: Are you sure this is a good choice? Compare with [BIP licensing](https://github.com/bitcoin/bips/blob/master/bip-0002.mediawiki#bip-licensing).

**Review by slowli — COMMENTED** [review]
Mostly good, but I think RFC licensing needs to be reworked - it may be too restrictive in certain scenarios.

File: README.md (hunk 1 above)
Comment (oudod) [inline README.md]: According to this comment https://github.com/rust-lang/rfcs/issues/1259#issuecomment-220753590, we are not first.

More info on `rust-lang/rfcs` licensing:
https://github.com/rust-lang/rfcs/blob/master/text/2044-license-rfcs.md
https://github.com/rust-lang/rfcs/pull/2075

File: README.md [hunk 6]
```diff
@@ -1 +1,245 @@
...
+## What the process is
+[What the process is]: #what-the-process-is
+
+In short, to get a major feature added to Exonum, one must first get the RFC
+merged into the RFC repository as a markdown file. At that point the RFC is
+"active" and may be implemented with the goal of eventual inclusion into Exonum.
+
+  - Fork the RFC repo [RFC repository]
+  - Copy `0000-template.md` to `text/0000-my-feature.md` (where "my-feature" is
+    descriptive. don't assign an RFC number yet).
```
Comment (Maria-Nosyk) [inline README.md]: > don't assign an RFC number yet

eliminate "an"; better "just yet"

File: README.md [hunk 7]
```diff
@@ -1 +1,245 @@
...
+
+Once an RFC becomes "active" then authors may implement it and submit the
+feature as a pull request to the Exonum repo. Being "active" is not a rubber
+stamp, and in particular still does not mean the feature will ultimately be
+merged; it does mean that in principle all the major stakeholders have agreed
+to the feature and are amenable to merging it.
+
+Furthermore, the fact that a given RFC has been accepted and is "active"
+implies nothing about what priority is assigned to its implementation, nor does
+it imply anything about whether a Exonum developer has been assigned the task of
```
Comment (Maria-Nosyk) [inline README.md]: > whether a Exonum developer has been assigned

a --> an

File: README.md [hunk 8]
```diff
@@ -1 +1,245 @@
...
+implementing the feature. While it is not *necessary* that the author of the
+RFC also write the implementation, it is by far the most effective way to see
```
Comment (Maria-Nosyk) [inline README.md]: > the author of the RFC also write the implementation

write --> writes

File: README.md [hunk 9]
```diff
@@ -1 +1,245 @@
...
+an RFC through to completion: authors should not expect that other project
```
Comment (Maria-Nosyk) [inline README.md]: > an RFC through to completion

an --> the

File: README.md [hunk 10]
```diff
@@ -1 +1,245 @@
...
+
+Modifications to "active" RFCs can be done in follow-up pull requests. We
+strive to write each RFC in a manner that it will reflect the final design of
+the feature; but the nature of the process means that we cannot expect every
+merged RFC to actually reflect what the end result will be at the time of the
+next major release.
+
+In general, once accepted, RFCs should not be substantially changed. Only very
+minor changes should be submitted as amendments. More substantial changes
+should be new RFCs, with a note added to the original RFC.
```
Comment (Maria-Nosyk) [inline README.md]: > should be new RFCs

better expand like "should be presented in the form of new RFCs"

File: README.md [hunk 11]
```diff
@@ -1 +1,245 @@
...
+
+## Reviewing RFCs
+[Reviewing RFCs]: #reviewing-rfcs
+
+While the RFC pull request is up, the shepherd may schedule meetings with the
+author and/or relevant stakeholders to discuss the issues in greater detail. In
+either case a summary from the meeting will be posted back to the RFC pull
+request.
+
+Maintainers of related repositories make final decisions about RFCs after the
```
Comment (Maria-Nosyk) [inline README.md]: > Maintainers of related repositories

The maintainers of the related repositories

File: README.md [hunk 12]
```diff
@@ -1 +1,245 @@
...
+benefits and drawbacks are well understood. These decisions can be made at any
+time, but the maintainers will regularly issue decisions. When a decision is made,
```
Comment (Maria-Nosyk) [inline README.md]: > When a decision is made,

the decision

File: README.md [hunk 13]
```diff
@@ -1 +1,245 @@
...
+the RFC pull request will either be merged or closed. In either case, if the
+reasoning is not clear from the discussion in thread, the maintainer of RFC
```
Comment (Maria-Nosyk) [inline README.md]: in thread --> in the thread
RFC repository --> the RFC repository

File: README.md [hunk 14]
```diff
@@ -1 +1,245 @@
...
+repository will add a comment describing the rationale for the decision.
+
+## Implementing an RFC
+[Implementing an RFC]: #implementing-an-rfc
+
+Some accepted RFCs represent vital features that need to be implemented right
+away. Other accepted RFCs can represent features that can wait until some
+arbitrary developer feels like doing the work. Every accepted RFC has an
+associated issue tracking its implementation in the Exonum repository; thus that
+associated issue can be assigned a priority via the triage process that the
```
Comment (Maria-Nosyk) [inline README.md]: > thus that associated issue -->

the associated issue

File: README.md [hunk 15]
```diff
@@ -1 +1,245 @@
...
+team uses for all issues in the Exonum repository.
+
+The author of an RFC is not obligated to implement it. Of course, the RFC
```
Comment (Maria-Nosyk) [inline README.md]: an RFC --> the RFC

File: README.md [hunk 16]
```diff
@@ -1 +1,245 @@
...
+"postponed" was used to postpone features until after 1.0. Postponed pull
+requests may be re-opened when the time is right.
+
+Usually an RFC pull request marked as "postponed" has already passed an
+informal first round of evaluation, namely the round of "do we think we would
+ever possibly consider making this change, as outlined in the RFC pull request,
+or some semi-obvious variation of it." (When the answer to the latter question
+is "no", then the appropriate response is to close the RFC, not postpone it.)
+
+## Help this is all too informal!
```
Comment (Maria-Nosyk) [inline README.md]: IMO "Help! This...." looks better

File: README.md [hunk 17]
```diff
@@ -1 +1,245 @@
...
+  - [Opening](#exonum-rfcs)
+  - [Table of Contents]
+  - [When you need to follow this process]
+  - [Before creating an RFC]
+  - [What the process is]
+  - [The RFC life-cycle]
+  - [Reviewing RFCs]
+  - [Implementing an RFC]
+  - [RFC Postponement]
+  - [Help this is all too informal!]
```
Comment (Maria-Nosyk) [inline README.md]: IMO "Help! This...." looks better

File: README.md [hunk 18]
```diff
@@ -1 +1,245 @@
...
+
+  - Changes in the public interface of the `exonum` crate.
+  - Architecture changes.
+  - Adding new features.
+  - Removing existing features.
+  - Changing feature behavior.
+
+Some changes do not require an RFC:
+
+  - Rephrasing, reorganizing, refactoring, or otherwise "changing shape does
```
Comment (Maria-Nosyk) [inline README.md]: otherwise --> "in other words"

File: README.md [hunk 19]
```diff
@@ -1 +1,245 @@
...
+
+  - Fork the RFC repo [RFC repository]
+  - Copy `0000-template.md` to `text/0000-my-feature.md` (where "my-feature" is
+    descriptive. don't assign an RFC number yet).
+  - Fill in the RFC. Put care into the details: RFCs that do not present
+    convincing motivation, demonstrate understanding of the impact of the
+    design, or are disingenuous about the drawbacks or alternatives tend to be
+    poorly-received.
+  - Submit a pull request. As a pull request the RFC will receive design
+    feedback from the larger community, and the author should be prepared to
```
Comment (Maria-Nosyk) [inline README.md]: > the larger community -->

a larger...

**Pull request merged**